spec/factories/invoices.rb

```diff
@@ -0,0 +1,7 @@
+FactoryGirl.define do
+  factory :invoice do
+    customer nil
```

Comment:
Looks good. You can actually leave off the nil on the relationships and factory girl still boogies.